src/Avalonia.Controls/ItemsControl.cs

```diff
@@ -249,8 +249,6 @@ protected virtual void OnContainersMaterialized(ItemContainerEventArgs e)
                         if (containerControl != null)
                         {
                             ((ISetLogicalParent)containerControl).SetParent(this);
-                            containerControl.SetValue(TemplatedParentProperty, null);
```

Comment:
This is no longer needed because it was clearing the inherited value. Now we have no inheritance.